lldb/source/Plugins/Process/gdb-remote/GDBRemoteCommunicationServerPlatform.cpp

```diff
@@ -160,66 +95,56 @@ GDBRemoteCommunicationServerPlatform::~GDBRemoteCommunicationServerPlatform() =
     default;
 
 Status GDBRemoteCommunicationServerPlatform::LaunchGDBServer(
-    const lldb_private::Args &args, std::string hostname, lldb::pid_t &pid,
-    std::optional<uint16_t> &port, std::string &socket_name) {
-  if (!port) {
-    llvm::Expected<uint16_t> available_port = m_port_map.GetNextAvailablePort();
-    if (available_port)
-      port = *available_port;
-    else
-      return Status(available_port.takeError());
-  }
-
-  // Spawn a new thread to accept the port that gets bound after binding to
-  // port 0 (zero).
+    const lldb_private::Args &args, lldb::pid_t &pid, std::string &socket_name,
+    shared_fd_t fd) {
+  std::ostringstream url;
+  if (fd == SharedSocket::kInvalidFD) {
+    if (m_socket_protocol == Socket::ProtocolTcp) {
+      // Just check that GDBServer exists. GDBServer must be launched after
+      // accepting the connection.
+      if (!GetDebugserverPath(nullptr))
+        return Status::FromErrorString("unable to locate debugserver");
+      return Status();
```

Comment:
It is inconvenient. It can be called from 
GDBRemoteCommunicationServerPlatform::Handle_qLaunchGDBServer()
and
lldb-platform.cpp, client_handle().
We can rename it to something like `CheckOrLaunchGDBServer()`.
BTW, Handle_qLaunchGDBServer() does not really launch the process now too in case of TCP protocol.

Reply:
I am still looking for an answer to this one. If it's only the `Handle_qLaunchGDBServer` that needs to *not* launch an lldb-server then I don't see why the check couldn't/shouldn't be performed there.